### ISSUE TITLE
Error on 'make'

### ISSUE THREAD

**Issue by suboyar** [Netgear-A6210-linux-driver/Netgear-A6210#25]
When compiling the driver with `make` it outputs an error. This is possible a duplicate of #21 but the author closed the issue without a solution.  
```
export DBGFLAGS

*** Building driver with debug messages ***

cp -f os/linux/Makefile.6 /home/sully/Downloads/Netgear-A6210/os/linux/Makefile
make -C /lib/modules/5.12.14-arch1-1/build DBGFLAGS=-DDBG M=/home/sully/Downloads/Netgear-A6210/os/linux modules
make[1]: Entering directory '/usr/lib/modules/5.12.14-arch1-1/build'
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/assoc.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/auth.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/auth_rsp.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/sync.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/sanity.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/rtmp_data.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/connect.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/wpa.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/sta_cfg.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../sta/sta.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/sta_ioctl.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_vht.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/vht.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/vht.c: In function ‘build_vht_op_ie’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/vht.c:405:54: warning: this statement may fall through [-Wimplicit-fallthrough=]
  405 |                         vht_op.basic_mcs_set.mcs_ss2 = VHT_MCS_CAP_7;
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/vht.c:406:9: note: here
  406 |         case 1:
      |         ^~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/crypt_md5.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/crypt_sha2.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/crypt_hmac.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/crypt_aes.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/crypt_arc4.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.c: In function ‘RTMPCheckHt’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.c:5199:53: warning: this statement may fall through [-Wimplicit-fallthrough=]
 5199 |                 pAd->MlmeAux.HtCapability.MCSSet[2] = 0xff;
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.c:5200:9: note: here
 5200 |         case 2:
      |         ^~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.c:5201:53: warning: this statement may fall through [-Wimplicit-fallthrough=]
 5201 |                 pAd->MlmeAux.HtCapability.MCSSet[1] = 0xff;
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/mlme.c:5202:9: note: here
 5202 |         case 1:
      |         ^~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wep.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/action.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_data.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/rtmp_init.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/rtmp_init_inf.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_tkip.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_aes.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.c: In function ‘BuildChannelList’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.c:100:41: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  100 |                                         if (pChannelList[i] == pAd->TxPower[j].Channel)
      |                                         ^~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.c:104:49: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  104 |                                                 pAd->ChannelList[index + i].Flags =
      |                                                 ^~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.c:205:41: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
  205 |                                         if (pChannelList[i] == pAd->TxPower[j].Channel)
      |                                         ^~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sync.c:209:49: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  209 |                                                 pAd->ChannelList[index + i].Flags =
      |                                                 ^~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/eeprom.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_sanity.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_info.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_cfg.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_cfg.c: In function ‘RT_CfgSetFixedTxPhyMode’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_cfg.c:605:37: warning: this statement may fall through [-Wimplicit-fallthrough=]
  605 |                         fix_tx_mode = value;
      |                         ~~~~~~~~~~~~^~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_cfg.c:606:17: note: here
  606 |                 default:
      |                 ^~~~~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wpa.o
In file included from ./include/linux/string.h:269,
                 from ./include/linux/bitmap.h:9,
                 from ./include/linux/cpumask.h:12,
                 from ./arch/x86/include/asm/cpumask.h:5,
                 from ./arch/x86/include/asm/msr.h:11,
                 from ./arch/x86/include/asm/processor.h:22,
                 from ./arch/x86/include/asm/timex.h:5,
                 from ./include/linux/timex.h:65,
                 from ./include/linux/time32.h:13,
                 from ./include/linux/time.h:60,
                 from ./include/linux/stat.h:19,
                 from ./include/linux/module.h:13,
                 from /home/sully/Downloads/Netgear-A6210/include/os/rt_linux.h:14,
                 from /home/sully/Downloads/Netgear-A6210/include/rtmp_os.h:30,
                 from /home/sully/Downloads/Netgear-A6210/include/rtmp_comm.h:64,
                 from /home/sully/Downloads/Netgear-A6210/include/rt_config.h:34,
                 from /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wpa.c:29:
In function ‘memmove’,
    inlined from ‘ConstructEapolKeyData’ at /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wpa.c:3742:3,
    inlined from ‘ConstructEapolMsg.constprop’ at /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wpa.c:3638:2:
./include/linux/fortify-string.h:21:33: warning: argument 2 null where non-null expected [-Wnonnull]
   21 | #define __underlying_memmove    __builtin_memmove
      |                                 ^
./include/linux/fortify-string.h:207:16: note: in expansion of macro ‘__underlying_memmove’
  207 |         return __underlying_memmove(p, q, size);
      |                ^~~~~~~~~~~~~~~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_wpa.c: In function ‘ConstructEapolMsg.constprop’:
./include/linux/fortify-string.h:21:33: note: in a call to built-in function ‘__builtin_memmove’
   21 | #define __underlying_memmove    __builtin_memmove
      |                                 ^
./include/linux/fortify-string.h:207:16: note: in expansion of macro ‘__underlying_memmove’
  207 |         return __underlying_memmove(p, q, size);
      |                ^~~~~~~~~~~~~~~~~~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_radar.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/spectrum.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/rtmp_timer.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/rt_channel.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_profile.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_asic.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/scan.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_cmd.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/ps.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/sys_log.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/txpower.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/ra_ctrl.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/ra_ctrl.c: In function ‘RTMPSetSupportMCS’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/ra_ctrl.c:3122:44: warning: this statement may fall through [-Wimplicit-fallthrough=]
 3122 |                                         if (vht_cap->mcs_set.rx_mcs_map.mcs_ss2 < VHT_MCS_CAP_NA)
      |                                            ^
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/ra_ctrl.c:3136:33: note: here
 3136 |                                 case 1:
      |                                 ^~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_legacy.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_legacy.c: In function ‘StaQuickResponeForRateUpExec’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_legacy.c:1382:33: warning: this ‘if’ clause does not guard... [-Wmisleading-indentation]
 1382 |                                 if (HwTxCnt)
      |                                 ^~
In file included from /home/sully/Downloads/Netgear-A6210/include/rtmp_os.h:30,
                 from /home/sully/Downloads/Netgear-A6210/include/rtmp_comm.h:64,
                 from /home/sully/Downloads/Netgear-A6210/include/rt_config.h:34,
                 from /home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_legacy.c:25:
/home/sully/Downloads/Netgear-A6210/include/os/rt_linux.h:582:1: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the ‘if’
  582 | do {                                            \
      | ^~
/home/sully/Downloads/Netgear-A6210/include/os/rt_linux.h:593:30: note: in expansion of macro ‘DBGPRINT_RAW’
  593 | #define DBGPRINT(Level, Fmt) DBGPRINT_RAW(Level, Fmt)
      |                              ^~~~~~~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_legacy.c:1385:41: note: in expansion of macro ‘DBGPRINT’
 1385 |                                         DBGPRINT(RT_DEBUG_INFO | DBG_FUNC_RA, ("%s():TxErrRatio(wcid:%d, MCS:%d, Hw:0x%lx-0x%lx, Sw:0x%lx-%lx)\n",
      |                                         ^~~~~~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_ags.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../chips/rtmp_chip.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_entrytb.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../tx_rx/wdev_tx.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_profile.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../phy/phy.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../phy/rf.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../phy/rlt_phy.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../mac/rtmp_mac.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../mac/ral_nmac.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../rate_ctrl/alg_grp.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/ba_action.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.c: In function ‘RTMPSetHT’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.c:280:43: warning: this statement may fall through [-Wimplicit-fallthrough=]
  280 |                         ht_cap->MCSSet[2] =  0xff;
      |                         ~~~~~~~~~~~~~~~~~~^~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.c:281:17: note: here
  281 |                 case 2:
      |                 ^~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.c:282:43: warning: this statement may fall through [-Wimplicit-fallthrough=]
  282 |                         ht_cap->MCSSet[1] =  0xff;
      |                         ~~~~~~~~~~~~~~~~~~^~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../mgmt/mgmt_ht.c:283:17: note: here
  283 |                 case 1:
      |                 ^~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_txbf.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_txbf_cal.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_txbf_cal.c: In function ‘mt76x2_ITxBFLNACalibration’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_txbf_cal.c:1919:17: warning: this statement may fall through [-Wimplicit-fallthrough=]
 1919 |                 ITxBFSetEEPROM(pAd, 0, &lnaParams, 0);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../common/cmm_txbf_cal.c:1921:9: note: here
 1921 |         case 2:
      |         ^~~~
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../common/rt_os_util.o
  CC [M]  /home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.o
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpOSFileOpen’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:674:17: error: implicit declaration of function ‘get_fs’; did you mean ‘sget_fc’? [-Werror=implicit-function-declaration]
  674 |         oldfs = get_fs();
      |                 ^~~~~~
      |                 sget_fc
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:674:17: error: incompatible types when assigning to type ‘mm_segment_t’ from type ‘int’
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:675:9: error: implicit declaration of function ‘set_fs’; did you mean ‘sget_fc’? [-Werror=implicit-function-declaration]
  675 |         set_fs(KERNEL_DS);
      |         ^~~~~~
      |         sget_fc
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:675:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
  675 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:675:16: note: each undeclared identifier is reported only once for each function it appears in
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpOSFileRead’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:705:27: error: invalid initializer
  705 |         mm_segment_t fs = get_fs();
      |                           ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:707:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
  707 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpOSFileWrite’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:723:30: error: invalid initializer
  723 |         mm_segment_t oldfs = get_fs();
      |                              ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:725:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
  725 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘__RtmpOSFSInfoChange’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:751:33: error: incompatible types when assigning to type ‘mm_segment_t’ from type ‘int’
  751 |                 pOSFSInfo->fs = get_fs();
      |                                 ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:752:24: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
  752 |                 set_fs(KERNEL_DS);
      |                        ^~~~~~~~~
      |                        KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpDrvAllMacPrint’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1615:19: error: incompatible types when assigning to type ‘mm_segment_t’ from type ‘int’
 1615 |         orig_fs = get_fs();
      |                   ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1616:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
 1616 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpDrvAllE2PPrint’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1663:19: error: incompatible types when assigning to type ‘mm_segment_t’ from type ‘int’
 1663 |         orig_fs = get_fs();
      |                   ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1664:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
 1664 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c: In function ‘RtmpDrvAllRFPrint’:
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1704:19: error: incompatible types when assigning to type ‘mm_segment_t’ from type ‘int’
 1704 |         orig_fs = get_fs();
      |                   ^~~~~~
/home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.c:1705:16: error: ‘KERNEL_DS’ undeclared (first use in this function); did you mean ‘KERNFS_NS’?
 1705 |         set_fs(KERNEL_DS);
      |                ^~~~~~~~~
      |                KERNFS_NS
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:271: /home/sully/Downloads/Netgear-A6210/os/linux/../../os/linux/rt_linux.o] Error 1
make[1]: *** [Makefile:1859: /home/sully/Downloads/Netgear-A6210/os/linux] Error 2
make[1]: Leaving directory '/usr/lib/modules/5.12.14-arch1-1/build'
make: *** [Makefile:65: debug] Error 2
```
It seems like there is an issue with using an undeclared value and a function; `get_fs()` should be `sget_fc` and `KERNEL_DS` should be `KERNFS_NS` in `os/linux/rt_linux.c`

### System information
- Arch Linux
- 5.12.14

**Comment by d-vesely** [comment]
I am having the same issue. I was working with a live USB boot of Pop_OS.

**Comment by toddcscar-prime** [comment]
Same issue on:
`
Linux version 5.11.0-43-generic (buildd@lcy02-amd64-036)
gcc version 9.3.0 (Ubuntu 9.3.0-17ubuntu1~20.04) 
`

**Comment by robinhood2014** [comment]
Same issue on Manjaro (Arch Linux) running kernel 5.15.12-1-MANJARO.

**Comment by genodeftest** [comment]
Closing due to inactivity. Please try the mainline kernel's `mt76` module instead, which should be automatically loaded if you use hardware that has been partially supported by this kernel driver module.
